modules/courses/public/views/pages/index.liquid

```diff
@@ -0,0 +1,35 @@
+---
+slug: course(/:id)
+layout: modules/courses/courses
+---
+{% liquid
+  function current_user = 'modules/user/lib/queries/user/current'
+
+  unless current_user.id
+    redirect_to '/user/register'
+  endunless
+
+  function can_manage_course = 'modules/permission/lib/helpers/can_do', requester: current_user, do: 'courses.manage'
+
+  assign status = '["published", "unpublished"]' | parse_json
+
+  graphql modules = 'modules/courses/module/search', withLessons: true, status: status
```

Comment:
best practice recommendation is to avoid invoking graphql directly in page and wrap them into lib/queries object, similarly how we wrap business logic into lib/commands